### PR TITLE
Simplify the implementation and understanding of the specification

### PR DESCRIPTION
**Pull request by rodweb**
fixed: #113

### CODE REVIEW

**Comment by rafaeldelboni** [comment]
I would add a file `.lsp/config.edn` with the following content:
```clojure
{:clean {:ns-inner-blocks-indentation :same-line}}
```

**Comment by rafaeldelboni** [comment]
Hello, I liked of the end result, the one thing I'm missing is a clear distinction if a namespace deals with IO(Impure functions) or not.
Usually in the Hexagonal architecture we have some default namespaces to define this like:
- Adapters, pure functions that adapts data structure, usually all public functions of this namespaces are unit tested.
  IE. The current ns in this PR `moclojer.specs.openapi` feels a lot like an adapter and we could rename it to `moclojer.adapters.openapi` or `moclojer.adapters.specs.openapi`.
- Logic, pure functions used to define the business logic, flow direction or build data structures based on logic 
  IE. Current ns `moclojer.router` feels a lot like this, we could rename to `moclojer.logic.router` as example.
- Ports/Diplomats impure functions that handle the extremities of the system, usually everything that deals with IO, Database, Http, Messaging
- Application/Controllers impure functions that orchestrate pure functions to impure functions usually a interesting http request data flow would be: port/http-in -> adapter -> controller -> logic -> controller -> port/db -> adapt -> controller -> port/http-in (response).

I'm not saying that you need to follow this and you don't need to fit all functions in those namespaces, but the best benefit of this architecture is the clear distinction of what to expect of a namespace before even opening the function and it eases how to test the functions, usually all the functions in pure namespace types are unit tested and impure functions are testes thought integration tests.

I have an example of this in a crud microservice:
https://github.com/parenthesin/microservice-boilerplate/tree/main/src/microservice_boilerplate

And there is a very good video about this subject:
https://www.youtube.com/watch?v=US8QG9I1XW0

I would like to hear @matheusfrancisco about this.

**Comment by matheusfrancisco** [comment]
@rafaeldelboni  great suggestions ❤️ .


@rodweb and I were talking about some function names and namespaces yesterday and I would think was because we didn´t have those distinctions. 
In my opinion, your suggestion is a great suggestion I would go in your direction because it will make our code clean and we will know what each namespace can have, as you said "without opening" 

about the suggestions, I would change the namespace `moclojer.specs.openapi `to `moclojer.adapters.openapi`
and also change `moclojer.router` to `moclojer.logic.router`.
what do you think @rodweb?
 
I will take a look at `https://github.com/parenthesin/microservice-boilerplate/tree/main/src/microservice_boilerplate`
your code because I realize Moclojer is a webserver also.

I will watch the video also.



Thanks @rafaeldelboni 🎊

**Comment by rodweb** [comment]
Hey @rafaeldelboni, thank you very much for your input. I'll definitely take your suggestions into account. I appreciate the references you linked as well.

@matheusfrancisco I'm leaning towards just making this PR ready for review before making more changes to the namespaces.

**Review by avelino — CHANGES_REQUESTED** [review]
route host is not working

```yaml
- endpoint:
    # Note: the method could be omitted because GET is the default
    method: GET
    path: /hello3/:username
    host: avelino.run
    response:
      # Note: the status could be omitted because 200 is the default
      status: 200
      headers:
        Content-Type: application/json
      # Note: the body will receive the value passed in the url using the
      # :username placeholder
      body: >
        {
          "hello3": "{{path-params.username}}!"
        }
```
```
curl http://127.0.0.1:8000/hello3/avelino
{
  "hello3": "avelino!"
}
```

it shouldn't work on host `127.0.0.1`, only on `avelino.run`

File: project.clj
Comment (matheusfrancisco) [inline project.clj:1]: @avelino  does this file need to be kept? it was an empty project.clj, is there any pip depends on this?

**Comment by rodweb** [comment]
@avelino should be working for moclojer spec now

```
❯ curl localhost:8000/multihost
Not Found%
❯ curl moclojer.com:8000/multihost
{
  "domain": "moclojer.com"
}
❯ curl sub.moclojer.com:8000/multihost-sub
{
  "domain": "sub.moclojer.com"
}
❯ curl sub.moclojer.com:8000/multihost
Not Found%
```

**Review by matheusfrancisco — APPROVED** [review]

**Review by matheusfrancisco — APPROVED** [review]

**Review by avelino — APPROVED** [review]
LGTM

**Pull request merged**